### PR TITLE
Fix the v78 migration script

### LINKED ISSUE #5759: Migration error: rename repo is_bare to repo is_empty
- Gitea version (or commit ref): docker: gitea/gitea:latest
- Git version: 2.18.1
- Operating system: Docker
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL (10.3.7-MariaDB)
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

I am running gitea inside docker container: `gitea/gitea:latest` - so I assume this is the latest version. After the last update with watchtower the container won't come up again. Checking the logfiles:

```
2019/01/18 09:07:18 [I] Migration: rename repo is_bare to repo is_empty
2019/01/18 09:07:18 [...itea/routers/init.go:61 GlobalInit()] [E] Failed to initialize ORM engine: migrate: do migrate: select repositories: Error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '"repository" RENAME COLUMN "is_bare" TO "is_empty"' at line 1
```

Not pretty sure but while `rename column` works for MySQL I think it is not working in MariaDB. `change column` should work for both DBs.

[MySQL Ref](https://dev.mysql.com/doc/refman/8.0/en/alter-table.html)
bildschirmfoto 2019-01-18 um 10 31 26

[MaraiDB Ref](https://mariadb.com/kb/en/library/alter-table/)
bildschirmfoto 2019-01-18 um 10 32 14

Working statement looks like:

```
ALTER TABLE
    `repository`
CHANGE COLUMN
    `is_bare` `is_empty` tinyint(1);
```

### PR DESCRIPTION
**Pull request by zeripath**
Unfortunately the last fix didn't completely fix the migration to v79 of the db
due to bug with schema locking during sess.Sync2. This should fix this issue.

Fix #5759, #5779

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

**Review by adelowo — APPROVED** [review]

**Comment by zeripath** [comment]
Guys before you merge if you could double check with postgres etc that this won't break them/will fix them too that would be excellent. It certainly fixes sqlite.

**Comment by zeripath** [comment]
Btw #5773 is the reason I noticed this and the reason I know that it definitely works on sqlite.

File: models/migrations/v78.go [hunk 1]
```diff
@@ -37,6 +38,17 @@ func renameRepoIsBareToIsEmpty(x *xorm.Engine) error {
 		return fmt.Errorf("Drop index failed: %v", err)
 	}
 
+	if err = sess.Commit(); err != nil {
+		return err
+	}
+
+	// Then reset the values
+	sess = x.NewSession()
+	defer sess.Close()
```
Comment (lunny) [inline models/migrations/v78.go:47]: I think this two lines is unnecessary but xorm should work.

Comment (zeripath) [reply models/migrations/v78.go:47]: Hiya @lunny, dropping the index and sync2 have to be in different transactions on sqlite as the schema becomes locked by sync2 which conflicts with the index drop. Therefore the original single transaction fails on sqlite and is rolled back.

**Comment by zeripath** [comment]
Thanks @techknowlogick.

File: models/migrations/v78.go (hunk 1 above)
Comment (lunny) [reply models/migrations/v78.go:47]: @zeripath I know. I mean
```Go
sess := x.NewSession()
defer sess.Close()

sess.Begin()
sess.Commit()

sess.Begin()
sess.Commit()
```
The second `NewSession` is uncessary.

**Pull request merged**